templates/conf.d/nginx.conf.erb

```diff
@@ -74,7 +74,11 @@ http {
   default_type  application/octet-stream;
 <% if @log_format -%>
 <% @log_format.sort_by{|k,v| k}.each do |key,value| -%>
-  log_format <%= key %> '<%= value %>';
+  <%- if value.is_a?(String) -%>
+  log_format <%= key %> <%= Puppet::Pops::Types::StringConverter.convert(value, '%p') %>;
```

Comment:
What's this string converter? Looks more like it's some private type.

Reply:
Good catch, I was at first trying to use EPP but this is a massive change that needs more time than I can afford ATM. `Puppet::Pops::Types::StringConverter` is what implements `String($value, '%p')` in the Puppet language, but since we are on the Ruby side, we can just `String#inspect` I suppose…

Reply:
Seems to do the right thing! I amended the commit accordingly.